File: Bridge/System/Collections/Generic/KeyValuePair.cs
using Bridge;

namespace System.Collections.Generic
{
    [External]
    [Immutable]
    [Reflectable]
    public struct KeyValuePair<TKey, TValue> : IBridgeClass
    {
        public extern KeyValuePair(TKey key, TValue value);

        [Convention(Notation.LowerCamelCase)] //[Field]
        public extern TKey Key
        {
            get;
        }

        [Convention(Notation.LowerCamelCase)] //[Field]
        public extern TValue Value
        {
            get;
        }

        [Convention(Notation.None)]
        public extern void Deconstruct(out TKey key, out TValue value);
    }
}

File: Bridge/Resources/Collections/KeyValuePair.js
    Bridge.define('System.Collections.Generic.KeyValuePair$2', function (TKey, TValue) {
        return {
            $kind: "struct",

            statics: {
                getDefaultValue: function () {
                    return new (System.Collections.Generic.KeyValuePair$2(TKey, TValue))(Bridge.getDefaultValue(TKey), Bridge.getDefaultValue(TValue));
                }
            },

            ctor: function (key, value) {
                if (key === undefined) {
                    key = Bridge.getDefaultValue(TKey);
                }

                if (value === undefined) {
                    value = Bridge.getDefaultValue(TValue);
                }

                this.$initialize();
                this.key = key;
                this.value = value;
            },

            Deconstruct: function (key, value) {
                key.v = this.key;
                value.v = this.value;
            },

            toString: function () {
                var s = "[";

                if (this.key != null) {
                    s += Bridge.toString(this.key);
                }

                s += ", ";

                if (this.value != null) {
                    s += Bridge.toString(this.value);
                }

                s += "]";

                return s;
            }
        };
    });